### PR TITLE
Implement all tf interpolation upscaling methods

### PR DESCRIPTION
**Pull request by Mahrkeenerh**
As mentioned in #16243 

This pull request implements all the available `tf.image.resize` methods to be used for the `UpSampling2D` layer.

### CODE REVIEW

**Review by fchollet — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: keras/layers/reshaping/up_sampling2d.py [hunk 1]
```diff
@@ -90,9 +90,19 @@ def __init__(self,
...
+      'bicubic': tf.image.ResizeMethod.BICUBIC,
+      'bilinear': tf.image.ResizeMethod.BILINEAR,
+      'gaussian': tf.image.ResizeMethod.GAUSSIAN,
+      'lanczos3': tf.image.ResizeMethod.LANCZOS3,
+      'lanczos5': tf.image.ResizeMethod.LANCZOS5,
+      'mitchellcubic': tf.image.ResizeMethod.MITCHELLCUBIC,
+      'nearest': tf.image.ResizeMethod.NEAREST_NEIGHBOR,
+    }
+    if interpolation not in interpolations:
+      raise ValueError('`interpolation` argument should be from `tf.image.ResizeMethod`. '
```
[the remark this answers is not in the document]
Comment (Mahrkeenerh) [reply keras/layers/reshaping/up_sampling2d.py]: Changed in [Implement review fixes](https://github.com/keras-team/keras/pull/16249/commits/895cf2090c665fac2a50549932acd55cd2dc3856)

File: keras/layers/reshaping/up_sampling2d.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ class UpSampling2D(Layer):
       It defaults to the `image_data_format` value found in your
       Keras config file at `~/.keras/keras.json`.
       If you never set it, then it will be "channels_last".
-    interpolation: A string, one of `nearest` or `bilinear`.
+    interpolation: A string any of `tf.image.ResizeMethod`.
```
[the remark this answers is not in the document]
Comment (Mahrkeenerh) [reply keras/layers/reshaping/up_sampling2d.py]: Changed in [Implement review fixes](https://github.com/keras-team/keras/pull/16249/commits/895cf2090c665fac2a50549932acd55cd2dc3856)

File: keras/backend.py [hunk 3]
```diff
@@ -3415,15 +3415,21 @@ def resize_images(x, height_factor, width_factor, data_format,
...
+    'mitchellcubic': tf.image.ResizeMethod.MITCHELLCUBIC,
+    'nearest': tf.image.ResizeMethod.NEAREST_NEIGHBOR,
+  }
+  if interpolation in interpolations:
+    x = tf.image.resize(x, new_shape, method=interpolations[interpolation])
   else:
-    raise ValueError('interpolation should be one '
-                     'of "nearest" or "bilinear".')
+    raise ValueError('interpolation should be '
+                     'from `tf.image.ResizeMethod`.')
```
[the remark this answers is not in the document]
Comment (Mahrkeenerh) [reply keras/backend.py]: Changed in [Implement review fixes](https://github.com/keras-team/keras/pull/16249/commits/895cf2090c665fac2a50549932acd55cd2dc3856)

File: keras/backend.py [hunk 4]
```diff
@@ -3389,7 +3389,7 @@ def resize_images(x, height_factor, width_factor, data_format,
       height_factor: Positive integer.
       width_factor: Positive integer.
       data_format: One of `"channels_first"`, `"channels_last"`.
-      interpolation: A string, one of `nearest` or `bilinear`.
+      interpolation: A string any of `tf.image.ResizeMethod`.
```
[the remark this answers is not in the document]
Comment (Mahrkeenerh) [reply keras/backend.py]: Changed in [Implement review fixes](https://github.com/keras-team/keras/pull/16249/commits/895cf2090c665fac2a50549932acd55cd2dc3856)

File: keras/backend.py [hunk 5]
```diff
@@ -3389,7 +3389,8 @@ def resize_images(x, height_factor, width_factor, data_format,
       height_factor: Positive integer.
       width_factor: Positive integer.
       data_format: One of `"channels_first"`, `"channels_last"`.
-      interpolation: A string, one of `nearest` or `bilinear`.
+      interpolation: A string, one of `area`, `bicubic`, `bilinear`,
```
Comment (fchollet) [inline keras/backend.py:3392]: To match the description of `data_format`, use:

```
A string, one of `"nearest"`, `"area"`, `"bicubic"`, `"bilinear"`,
`"gaussian"`, `"lanczos3"`, `"lanczos5"`, `"mitchellcubic"`.
Defaults to `"nearest"`.
```

File: keras/backend.py [hunk 6]
```diff
@@ -3415,15 +3416,22 @@ def resize_images(x, height_factor, width_factor, data_format,
...
+    'area': tf.image.ResizeMethod.AREA,
+    'bicubic': tf.image.ResizeMethod.BICUBIC,
+    'bilinear': tf.image.ResizeMethod.BILINEAR,
+    'gaussian': tf.image.ResizeMethod.GAUSSIAN,
+    'lanczos3': tf.image.ResizeMethod.LANCZOS3,
+    'lanczos5': tf.image.ResizeMethod.LANCZOS5,
+    'mitchellcubic': tf.image.ResizeMethod.MITCHELLCUBIC,
+    'nearest': tf.image.ResizeMethod.NEAREST_NEIGHBOR,
+  }
+  interploations_list = '`' + '`, `'.join(interpolations.keys()) + '`'
```
Comment (fchollet) [inline keras/backend.py:3429]: Replace \` with "

File: keras/layers/reshaping/up_sampling2d.py [hunk 7]
```diff
@@ -65,7 +65,8 @@ class UpSampling2D(Layer):
       It defaults to the `image_data_format` value found in your
       Keras config file at `~/.keras/keras.json`.
       If you never set it, then it will be "channels_last".
-    interpolation: A string, one of `nearest` or `bilinear`.
+    interpolation: A string, one of `area`, `bicubic`, `bilinear`, `gaussian`,
```
Comment (fchollet) [inline keras/layers/reshaping/up_sampling2d.py:68]: Likewise

File: keras/layers/reshaping/up_sampling2d.py [hunk 8]
```diff
@@ -90,9 +91,20 @@ def __init__(self,
...
+      'area': tf.image.ResizeMethod.AREA,
+      'bicubic': tf.image.ResizeMethod.BICUBIC,
+      'bilinear': tf.image.ResizeMethod.BILINEAR,
+      'gaussian': tf.image.ResizeMethod.GAUSSIAN,
+      'lanczos3': tf.image.ResizeMethod.LANCZOS3,
+      'lanczos5': tf.image.ResizeMethod.LANCZOS5,
+      'mitchellcubic': tf.image.ResizeMethod.MITCHELLCUBIC,
+      'nearest': tf.image.ResizeMethod.NEAREST_NEIGHBOR,
+    }
+    interploations_list = '`' + '`, `'.join(interpolations.keys()) + '`'
```
Comment (fchollet) [inline keras/layers/reshaping/up_sampling2d.py:104]: Likewise

**Review by fchollet — COMMENTED** [review]
Thanks for the update!

File: keras/layers/reshaping/up_sampling2d.py (hunk 7 above)
Comment (Mahrkeenerh) [reply keras/layers/reshaping/up_sampling2d.py:68]: So should this one be `"area"` or "area" ?

Comment (Mahrkeenerh) [reply keras/layers/reshaping/up_sampling2d.py:68]: @fchollet

File: keras/backend.py (hunk 5 above)
Comment (Mahrkeenerh) [reply keras/backend.py]: Changed in [Fix string formatting](https://github.com/keras-team/keras/pull/16249/commits/aa8af10a160845df30a8b28470e866e02c68bdf5)

File: keras/backend.py (hunk 6 above)
Comment (Mahrkeenerh) [reply keras/backend.py]: Changed in [Fix string formatting](https://github.com/keras-team/keras/pull/16249/commits/aa8af10a160845df30a8b28470e866e02c68bdf5)

File: keras/layers/reshaping/up_sampling2d.py (hunk 7 above)
Comment (Mahrkeenerh) [reply keras/layers/reshaping/up_sampling2d.py]: Changed in [Fix string formatting](https://github.com/keras-team/keras/pull/16249/commits/aa8af10a160845df30a8b28470e866e02c68bdf5)

File: keras/layers/reshaping/up_sampling2d.py (hunk 8 above)
Comment (Mahrkeenerh) [reply keras/layers/reshaping/up_sampling2d.py]: Changed in [Fix string formatting](https://github.com/keras-team/keras/pull/16249/commits/aa8af10a160845df30a8b28470e866e02c68bdf5)

**Review by fchollet — APPROVED** [review]
LGTM, thank you

**Pull request merged**